packages/editor/src/core/components/nice-dropdown.tsx

```diff
@@ -0,0 +1,123 @@
+
+  const triggerRef = useRef<HTMLButtonElement>(null)
+
+  const handleChange = (newValue: string) => {
+    setSelectedValue(newValue)
+    onChange?.(newValue)
+    setIsConfirmed(true)
+  }
+
+  useEffect(() => {
+    if (triggerRef.current) {
+      setTriggerWidth(triggerRef.current.offsetWidth)
+    }
+  }, [])
+
+  useEffect(() => {
+    if (isOpen && !hasOpened) {
+      setHasOpened(true)
+    } else if (!isOpen && hasOpened) {
+      setIsConfirmed(true)
+    }
+  }, [isOpen, hasOpened])
+
+  // Update selectedValue if defaultValue changes
+  useEffect(() => {
+    setSelectedValue(defaultValue)
+  }, [defaultValue])
+
+  return (
+    <label
+      className={`h-100 mx-auto mb-0 mt-5 flex items-center text-almost-black ${className}`}
+    >
+      <span className="serlo-tooltip-trigger flex w-1/4 justify-between pr-[15px]">
+        {label}
+        <FaIcon className="ml-2" icon={faQuestionCircle} />
+        <EditorTooltip text={helpText} />
+      </span>
+      <span className="flex w-1/2 flex-row border-2 border-transparent">
+        <Select.Root
+          onValueChange={handleChange}
+          defaultValue={defaultValue}
+          onOpenChange={setIsOpen}
+          value={selectedValue}
+        >
+          <Select.Trigger
+            ref={triggerRef}
+            onKeyDown={(e) => e.stopPropagation()}
+            className={cn(
+              'serlo-input-font-reset rounded-md border-2 bg-editor-primary-100 px-2 text-left focus:border-2 focus:border-editor-primary',
+              'flex w-full flex-row justify-between',
+              isOpen
+                ? 'rounded-b-none border-b-0 border-editor-primary'
+                : 'border-editor-primary-100'
+            )}
+            style={{ height: '1.75em' }}
+          >
+            <Select.Value />
+            <Select.Icon className="ml-2 text-almost-black">
+              <FaIcon icon={isOpen ? faCaretUp : faCaretDown} />
+            </Select.Icon>
+          </Select.Trigger>
+          <Select.Content
+            className="mx-auto mt-[-2px] rounded-md rounded-t-none border-2 border-editor-primary bg-white"
+            side="bottom"
+            position="popper"
+            onKeyDown={(e) => e.stopPropagation()}
+            style={{ width: triggerWidth, borderTop: 'none' }}
+          >
+            <Select.Viewport>
+              {options.map(({ label, value }) => (
+                <Select.Item
+                  key={value.toString()}
+                  value={value.toString()}
+                  className="my-0 px-3 hover:bg-editor-primary-100 focus:bg-editor-primary-100"
+                >
+                  <Select.ItemText>{label}</Select.ItemText>
+                </Select.Item>
+              ))}
+            </Select.Viewport>
+          </Select.Content>
+        </Select.Root>
+      </span>
+      {isConfirmed && (
+        <span className="pointer-events-none ml-2 inline-flex items-center pr-3 text-green-500">
+          &#10003;
```

Comment:
optional: I usually just use the char directly (✓) or add a comment for readability

Reply:
aside: in most contexts this confirmed logic is not needed so we should probably have a prop in the comp to activate it.

Reply:
Added